### PR TITLE
Update Developer Guide

### PR DESCRIPTION
**Pull request by domlimm**
- Update add command format
- Refactor add command title to use plural form
- Change user story table format from IntelliJ's suggestion
- Fix program name in value proposition

More info #45.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -257,14 +257,14 @@ _{Explain here how the data archiving feature will be implemented}_
 
 **Target user profile**:
 
-* Professor in charge of CS2103
+* Professor in from NUS School of Computing
```
Comment (lzan98) [inline docs/DeveloperGuide.md:260]: Perhaps a minor error here - an extra "in"

**Review by lzan98 — CHANGES_REQUESTED** [review]
LGTM! 😄 
Just a minor error in line 260 of DeveloperGuide.md

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (domlimm) [reply docs/DeveloperGuide.md:260]: Good catch!

**Review by lzan98 — APPROVED** [review]
👍 Thanks Dom!

**Comment by domlimm** [comment]
@lzan98 no thank you!!! 😁

**Pull request merged**